### PR TITLE
Es pja add new item

### LINKED ISSUE #6: 4. As a user, I want to add a new item to my shopping list
A shopping list item consists of the following data points:

* Name of item
* How soon are you likely to buy it again?
  * Soon (in the next 7 days)
  * Kind of soon (in the next 14 days)
  * Not soon (in the next 30 days)
* Last purchased date

AC:

* User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of soon, Not soon)
* When the user submits the form, the item is saved to the database, associated with the user’s token
* Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of soon”, and 30 for “Not soon”
* The last purchased date should be set to `null` initially (i.e. you can create an item without purchasing it)

### PR DESCRIPTION
**Pull request by erostribe**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Related Issue

Closes #6 

## Acceptance Criteria

- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of soon, Not soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of soon”, and 30 for “Not soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

Image 4-17-20 at 4 37 PM (1)

### After

Image 4-17-20 at 4 29 PM

## Testing Steps / QA Criteria

1)  Git pull the branch using `git pull origin es-pja-add-new-item` and `git checkout es-pja-add-new-item` to switch to the branch.
2) Run `npm install` to add dependencies, then `npm start` to run the app.
3) When the app runs in the browser, you will be presented with a page titled "Shopping List" with two buttons "Shopping" and "Add Item."
4) Clicking "Add Item" should present a form with a "Name" input, a Schedule option with a dropdown menu and 3 choices, and a Submit button.

### CODE REVIEW

File: src/components/AddItemForm.js
Comment (skillitzimberg) [inline src/components/AddItemForm.js]: I think this `prettier` import might be what's causing the checks to fail. I'm not sure. I did run `npm install` and I still get these messages in the terminal. Commenting out this import fixed it.

```
./src/components/AddItemForm.js
 Line 2:10:  'format' is defined but never used  no-unused-vars

./node_modules/prettier/parser-typescript.js
Module not found: Can't resolve '@microsoft/typescript-etw' in '/Users/ForeignFood/Development/collab_lab/tcl-6-smart-shopping-list/node_modules/prettier'
```

File: src/components/AddItemForm.js
Comment (skillitzimberg) [inline src/components/AddItemForm.js]: I couldn't find a record of it the ticket or AC, but is there supposed to be a default value for `next_purchase`?

For AC `next_purchase` should have an integer saved with the item. You might need an `onChange={this.handleNextPurchaseChange}` similar to the one you have for the item name.

File: src/components/AddItemForm.js
Comment (skillitzimberg) [inline src/components/AddItemForm.js]: Tyypo . . . : ) `next_purchase: null,`

File: src/components/AddItemForm.js
Comment (skillitzimberg) [inline src/components/AddItemForm.js]: `this.state.schedule` should be `this.state.next_purchase`

File: src/components/AddItemForm.js
Comment (SaraSweetie) [inline src/components/AddItemForm.js]: The submit button should be at the end of the form. You have a text field then the submit button, followed by two more form inputs. That is not good for accessibility.

File: src/components/AddItemForm.js
Comment (SaraSweetie) [reply src/components/AddItemForm.js]: The `next_purchase` state is not updating and saving properly to the firestore database. Your item name is updating properly. To have the select drop-down update the proper numerical value to the database, there are a couple of things you need to do.

```
this.handleSchdeule = this.handleSchdeule.bind(this);

  handleSchdeule(event) {
    this.setState({ 
      next_purchase: event.target.value,
     });
  }
// in the form
          <select value={this.state.value}
            onChange={this.handleSchdeule}> // add this like you have on the text field
            <option value="7">Soon</option>
            <option value="14">Kind Of Soon</option>
            <option value="30">Not Soon</option>
          </select>
```

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:57]: ❤️ this semantic and accessible form 😍

Alternatively, you could remove the `<br />` tags and use css:

```css
#addItemForm label {
  display: block;
}
```

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js]: 🚨 React wants `className` rather than `class`. `class` is a keyword in JS so we can't use it in `jsx`.

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js]: We can probably omit this value. We'll set this value in a later story when we mark that the item has been purchased.

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:13]: Not a blocker, just FYI in case you didn't know. With create-react-apps (CRA), we have a babel feature that will let you define method without [explicitly binding](https://reactjs.org/docs/react-without-es6.html#autobinding) `this` in the constructor.

```jsx
handleChange = event => {
  this.setState({ name: event.target.value });
}
```

Like the docs mention, it's an experimental feature so your mileage may vary. But it's definitely a common pattern.

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [reply src/components/AddItemForm.js]: You could make the state reset more consistent by defining an `initialState` variable. 

```jsx
const initialState = {
  name: 'some default or blank'
}

class Form extends Component {
  state = initialState

  handleSubmit = event => {
    ...submitLogic
    this.setState(initialState);
  }
}
```

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:60]: Double ❤️❤️ for using `<select>` 👏👏👏

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [reply src/components/AddItemForm.js]: The `select` element expects to have _some_ value passed to it. `null` throws an error: 



I would make the default `14` (kinda soon).

File: src/CSS/AddItemForm.css
Comment (stevelikesmusic) [inline src/CSS/AddItemForm.css:1]: A common practice with React apps is to keep component stylesheets adjacent to the components that consume them. What would you think of moving this file next to `AddItemForm.js`?

File: src/App.js
Comment (stevelikesmusic) [inline src/App.js]: 🚨 Why make the path `/add/form`? Did you have another view in mind for `/add`? If not, I would keep `/add` for the add form.

File: src/components/TestHistory.js
Comment (stevelikesmusic) [inline src/components/TestHistory.js:1]: What is this component for? Could it be deleted?

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:28]: Not at all a requirement for this ticket—just a question: How might you redirect the user back to their list after they successfully add an item?

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:3]: Now that #5 is merged, you could add `useToken` to this component and utilize the same token that the user has created and saved to `localStorage`.

File: .vscode/settings.json
Comment (stevelikesmusic) [inline .vscode/settings.json:2]: If we don't want to save this file to the repo, you can add it to `.gitignore`.

File: src/components/AddItemForm.js
Comment (stevelikesmusic) [inline src/components/AddItemForm.js:23]: We will expect to use this value as a number later on to calculate the next purchase date. What data type is `event.target.value` here? Is it a number? If not, how could we make it one?

**Review by stevelikesmusic — APPROVED** [review]
This is some great work @espain16 @erostribe 👏 

There are only a couple things I would add/change before merging:
1. Set a default for the `select` element that is not `null`
1. Use `className` instead of `class` on the form
1. Use `useToken` now that we have it, so we save the item with the list token the user created.

The other comments are suggestions or 🎉 Lot's of good stuff here!!

**Comment by stevelikesmusic** [comment]
One other comment:

I like the before/after in the PR details—I might add an additional step in the testing steps talking about what to expect once we submit the form. Let's get in the habit of writing a thorough description (especially about what you did and why you made the choices you did). You should be able to use these PRs as a resource showing how you communicate with a remote team. 

Again, lot's of great work 👏👏👏

File: src/components/AddItemForm.js
Comment (segdeha) [inline src/components/AddItemForm.js]: Sorry I didn't mention this in our Slack thread, but I believe you want to store the user's token along with the other data points, correct?

File: src/components/AddItemForm.js
Comment (segdeha) [inline src/components/AddItemForm.js]: Connected to line 32 above, store the token along with the other data

**Review by segdeha — APPROVED** [review]
Approving so that once you add the token as the other data point you're free to merge 💪

File: .vscode/settings.json
Comment (segdeha) [reply .vscode/settings.json:2]: Like Steve said, we should exclude this file as it's a code editor configuration

**Review by segdeha — APPROVED** [review]
This looks good 👍

**Review by SaraSweetie — APPROVED** [review]

**Pull request merged**
